app/src/shared/components/Prop/index.jsx

```diff
@@ -0,0 +1,15 @@
+// @flow
+
+import { useState, type Node } from 'react'
+
+type Props = {
+    initialValue?: any,
+    children: (any, (any) => void) => Node,
+}
+
+const Prop = ({ children, initialValue }: Props) => {
+    const [value, setValue] = useState(initialValue)
+    return children(value, setValue)
+}
```

Comment:
Cool, this enables useState in class components, but IMO… uh, very confusing name since it doesn't have anything to do with props, rather it's to do with state. `<UseState>` perhaps more immediately understandable?

Also strongly suggest only using this when a normal `useState` hook won't work i.e. only use in class components?